test/unit-tests/marginCalculatorNakedMargin.test.ts

```diff
@@ -0,0 +1,848 @@
+      )
+
+      // console.log('Expected margin: ', expectedRequiredNakedMargin.toString())
+      // console.log('Contract margin: ', requiredMargin.toString())
+      assert.equal(
+        requiredMargin.dividedBy(10 ** wethDecimals).toNumber(),
+        expectedRequiredNakedMargin,
+        'Required naked margin for put mismatch',
+      )
+
+      assert.isAtMost(
+        calcRelativeDiff(new BigNumber('0.1677778677'), requiredMargin.dividedBy(10 ** wethDecimals)).toNumber(),
+        errorDelta,
+        'big error delta',
+      )
+    })
+
+    it('should return required margin for naked margin vault: 100k options 2500$ WETH call option with 1800 spot price and 1 week to expiry', async () => {
+      // set product shock value
+      const spotShockValue = scaleNum(0.75, 27)
+      await calculator.setSpotShock(weth.address, usdc.address, weth.address, false, spotShockValue, {from: owner})
+
+      const shortAmount = 100000
+      const shortStrike = 2500
+      const underlyingPrice = 1800
+      const scaledShortAmount = scaleBigNum(shortAmount, 8)
+      const scaledShortStrike = scaleBigNum(shortStrike, 8)
+      const scaledUnderlyingPrice = scaleBigNum(underlyingPrice, 8)
+      const isPut = false
+      const optionExpiry = new BigNumber(await time.latest()).plus(timeToExpiry[0])
+      // get option upper bound value
+      const upperBoundValue = new BigNumber(
+        await calculator.findUpperBoundValue(weth.address, usdc.address, weth.address, false, optionExpiry),
+      )
+        .dividedBy(1e27)
+        .toNumber()
+
+      // console.log('Upper boudn value found: ', upperBoundValue.toString())
+      // expected required margin
+      const expectedRequiredNakedMargin = expectedRequiredMargin(
+        shortAmount,
+        shortStrike,
+        underlyingPrice,
+        isPut,
+        upperBoundValue,
+        productSpotShockValue.dividedBy(1e27).toNumber(),
+      )
+
+      const requiredMargin = new BigNumber(
+        await calculator.getNakedMarginRequired(
+          weth.address,
+          usdc.address,
+          weth.address,
+          scaledShortAmount,
+          scaledShortStrike,
+          scaledUnderlyingPrice,
+          optionExpiry,
+          wethDecimals,
+          isPut,
+        ),
+      )
+
+      // console.log('Expected margin: ', expectedRequiredNakedMargin.toString())
+      // console.log('Contract margin: ', requiredMargin.toString())
+      assert.equal(
+        requiredMargin.dividedBy(10 ** wethDecimals).toNumber(),
+        expectedRequiredNakedMargin,
+        'Required naked margin for put mismatch',
+      )
+
+      // assert.isAtMost(
+      //   calcRelativeDiff(new BigNumber('16777.78677'), requiredMargin.dividedBy(10 ** wethDecimals)).toNumber(),
+      //   errorDelta,
+      //   'big error delta',
+      // )
+    })
+
+    it('should return required margin for naked margin vault: 100$ WETH put option with 150 spot price and 2 weeks to expiry', async () => {
```

Comment:
could you also test it calculates it right for ITM calls and ITM puts? :D

Reply:
Ah u meant ITM instead of OTM, I guess it is the same calculation, IMO this cover the basics and will add more in integration tests.